tests/rgbpp/xudt/btc-transfer-all/1-btc-transfer-all.ts

```diff
@@ -0,0 +1,105 @@
+import { bitcoin } from 'rgbpp/btc';
+import { buildRgbppTransferAllTxs, sendRgbppTxGroups } from 'rgbpp';
+import { btcDataSource, isMainnet, collector, btcAccount } from '../../env';
+import { signPsbt } from '../../../../examples/rgbpp/shared/btc-account';
+import { saveCkbVirtualTxResult } from '../../../../examples/rgbpp/shared/utils';
+import { readStepLog } from '../../shared/utils';
+
+interface TestParams {
+  xudtTypeArgs: string;
+  fromAddress: string;
+  toAddress: string;
+}
+
+const rgbppTransferAllTxs = async ({ xudtTypeArgs, fromAddress, toAddress }: TestParams) => {
+  const result = await buildRgbppTransferAllTxs({
+    ckb: {
+      xudtTypeArgs,
+      collector,
+    },
+    btc: {
+      assetAddresses: [fromAddress],
+      fromAddress: fromAddress,
+      toAddress: toAddress,
+      dataSource: btcDataSource,
+      feeRate: 5,
+    },
+    isMainnet,
+  });
+
+  console.log('result.transactions.length', result.transactions.length);
+  console.log('result.summary.included.assets', result.summary.included.xudtAssets);
+  console.log('result.summary.excluded.assets', result.summary.excluded.xudtAssets);
+
+  const signedGroups = await Promise.all(
+    result.transactions.map(async (group) => {
+      const psbt = bitcoin.Psbt.fromHex(group.btc.psbtHex);
+
+      // Sign transactions
+      signPsbt(psbt, btcAccount);
+
+      psbt.finalizeAllInputs();
+
+      return {
+        ckbVirtualTxResult: JSON.stringify(group.ckb.virtualTxResult),
+        btcTxHex: psbt.extractTransaction().toHex(),
+      };
+    }),
+  );
+
+  const signedGroupsData = JSON.parse(JSON.stringify(signedGroups, null, 2));
+
+  // Save signedGroupsData
+  saveCkbVirtualTxResult(signedGroupsData, '1-btc-transfer-all');
+
+  console.log('signedGroups', signedGroupsData);
+
+  // Send transactions
+  const sentGroups = await sendRgbppTxGroups({
+    txGroups: signedGroups,
+    btcService: btcDataSource.service,
+  });
+  console.log('sentGroups', JSON.stringify(sentGroups, null, 2));
+
+  const successfulTxIds = sentGroups
+    .filter((group) => group.btcTxId)
+    .map((group) => `https://mempool.space/testnet/tx/${group.btcTxId}`);
+
+  console.log('Successful Transactions:', successfulTxIds.join('\n'));
+
+  try {
+    for (const group of sentGroups) {
```

Comment:
It is recommended to fetch all RGB++ transaction states with `promise.all` in one interval block instead of a loop